DependencyInjection/Compiler/RelationProviderPass.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * This file is part of the HateoasBundle package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Bazinga\Bundle\HateoasBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * @author Daniel Leech <daniel@dantleech.com>
+ */
+class RelationProviderPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $registryDefinition = $container->getDefinition('hateoas.configuration.provider.resolver.chain');
+
+        $relationProviderDefinitions = array();
+        foreach (array_keys($container->findTaggedServiceIds('hateoas.relation_provider')) as $id) {
+            $definition = $container->getDefinition($id);
+            $this->assertProvider($container, $definition);
+            $relationProviderDefinitions[] = $definition;
+        }
+
+        $registryDefinition->setArguments($relationProviderDefinitions);
```

Comment:
This is wrong, it should be `->setArguments([$relationProviderDefinitions]);` or `->replaceArgument(0, $relationProviderDefinitions);`